utils/httpUtils.ts

```diff
@@ -40,15 +60,35 @@ export const get = async (url: string, args: any, context: any) => {
 export const getFullResponse = async (url: string, args: any, context: any) => {
   try {
     const baseURL = process.env.API_URL;
-    const urlPrefix = args.snapshotLinkId ? `/pub/${args.snapshotLinkId}` : "";
-    const response = await fetch(baseURL + urlPrefix + url, {
-      method: "GET",
-      headers: {
-        Cookie: context.request.headers.get("cookie"),
-        "Content-Type": "application/json",
-      },
-    });
-    return response;
+    const nextGenEnabled = await isNextGenEnabled(context);
+    if (nextGenEnabled) {
+      // next gen details
+      const czidServicesToken = await getEnrichedToken(context);
+      const query = context.params.query;
+      const response = await fetch(process.env.API_URL + url, {
+        method: "POST",
+        headers: {
+          Cookie: context.request.headers.get("cookie"),
+          "Content-Type": "application/json",
+          "X-CSRF-Token": args?.input?.authenticityToken,
+          Authorization: `Bearer ${czidServicesToken}`,
+        },
+        body: JSON.stringify(query),
+      });
+      return response;
```

Comment:
would it be viable to put this code in the `callGraphqlSource` util function instead, and call that util function here?  This util file is getting fairly long and complex, so it'd be nice to extract the chunks that make sense.

https://github.com/chanzuckerberg/czid-graphql-federation-server/blob/010967bd3011dd1544a6bad984f4e64e2f3f0b08/utils/callGraphqlSource.ts#L3